boggle.js

```diff
@@ -6,26 +25,136 @@ function newBoggleBoard() {
     ['_', '_', '_', '_'],
   ];
 }
-
-/**
- * Prints out a boggle board.
- */
 function printBoggleBoard(boggleBoard) {
+  let isQ = false;
+  for (let i = 0; i < boggleBoard.length; i++) {
+    for (let j = 0; j < boggleBoard[i].length; j++) {
+      if (boggleBoard[i][j] === 'Q') {
+        boggleBoard[i][j] = 'Qu';
+        isQ = true;
+      }
+    }
+  }
+
+  if (isQ === true) {
+    for (let i = 0; i < boggleBoard.length; i++) {
+      for (let j = 0; j < boggleBoard[i].length; j++) {
+        boggleBoard[i][j] = boggleBoard[i][j].padEnd(2);
+      }
+    }
+  }
   for (let row of boggleBoard) {
     console.log(row.join(' '));
   }
 }
 
-/**
- * Shake a boggle board and fill it with letters.
- */
-function shake(boggleBoard) {
-  // This is your job. :)
+// function shake(boggleBoard) {
+// let abc = 'abcdefghijklmnopqrstuvwxyz';
+//   for(let row of boggleBoard){
+//     for(let i = 0; i < boggleBoard.length; i++){
+//       row[i] = abc[Math.floor(Math.random()*26)];
+//     }
+
+//   }
+//   return boggleBoard;
+
+// }
+function shake (boggleBoard) {
+  let dice = [
+    {
+      combo: 'AAEEGN',
+    },
+    {
+      combo: 'ELRTTY',
+    },
+    {
+      combo: 'AOOTTW',
+    },
+    {
+      combo: 'ABBJOO',
+    },
+    {
+      combo: 'EHRTVW',
+    },
+    {
+      combo: 'CIMOTU',
+    },
+    {
+      combo: 'DISTTY',
+    },
+    {
+      combo: 'EIOSST',
+    },
+    {
+      combo: 'DELRVY',
+    },
+    {
+      combo: 'ACHOPS',
+    },
+    {
+      combo: 'HIMNQU',
+    },
+    {
+      combo: 'EEINSU',
+    },
+    {
+      combo: 'EEGHNW',
+    },
+    {
+      combo: 'AFFKPS',
+    },
+    {
+      combo: 'HLNNRZ',
+    },
+    {
+      combo: 'DEILRX',
+    },
+
+  ];
+  for (let row of boggleBoard) {
+    for (let i = 0; i < boggleBoard.length; i++) {
+      let randomDice = Math.floor(Math.random() * 15);
+      let randomLetter = Math.floor(Math.random() * 5);
+      row[i] = dice[randomDice].combo[randomLetter];
+    }
+  }
   return boggleBoard;
 }
 
+function boardCheck() {
+  let array = [];
+  for (let i = 0; i < 4; i++) {
+    for (let j = 0; j < 4; j++) {
+      let str = '';
+      marked[i][j] = true;
+      explore(board, marked, i, j, str, array);
+      marked[i][j] = false;
+    }
+  }
+  return array;
+}
+
+function explore(board, marked, row, col, str, array) {
+  str += board[row][col];
+  if (str.length >= 3 & dict.hasOwnProperty(str)) {
+    array.push(str);
+  }
+  for (let i = row - 1; i <= row + 1; i++) {
+    for (let j = col - 1; j <= col + 1; j++) {
+      if (i >= 0 && j >= 0 && i < 4 && j < 4 && !marked[i][j]) {
+        marked[i][j] = true;
+        explore(board, marked, i, j, str, array);
```

Comment:
Nice! The recursive approach is much nicer for the real boggle rules.

For the only check horizontal/vertical it's a bit overkill. We can talk about how to structure the problem recursively, though. Find me today.